### PR TITLE
httpbakery: allow custom error responses

### PR DESCRIPTION
**Pull request by rogpeppe**
We have a need to deal with non-httpbakery-like error messages
so we allow that freedom in httpbakery.Client.Do.

### CODE REVIEW

**Comment by makyo** [comment]
:+1:

File: httpbakery/client.go [hunk 1]
```diff
@@ -213,13 +213,33 @@ func relativeURL(base, new string) (*url.URL, error) {
...
+// a non-nil Error if the response holds an error that
+// can be converted to an Error value. The getError call must leave the
+// response body unchanged otherwise.
+//
+// If getError is nil, DefaultGetError will be used.
+//
+// This method can be useful when dealing with APIs that
+// return their errors in a format incompatible with Error, but the
+// need for it should be avoided when creating new APIs,
+// as it makes the endpoints not amenable to generic tools.
```
Comment (mhilton) [inline httpbakery/client.go]: "less" as opposed to "not"?

**Comment by mhilton** [comment]
:+1:

File: httpbakery/client.go (hunk 1 above)
Comment (rogpeppe) [inline httpbakery/client.go]: done

**Pull request merged**